CONVENTIONS.md

```diff
@@ -290,6 +290,24 @@ Resource request review history:
 * [BZ 1920159](https://bugzilla.redhat.com/show_bug.cgi?id=1920159) --
   for tracking changes in 4.7/4.8 for single-node RAN
 
+##### Allowed use of limits
+
+There is an exception process for adding limits to payload workloads. In the following
+scenarios workloads may be given permission to set limits:
+
+* Workloads that are scale-invariant and have a fixed memory or CPU cost no matter what the scale of the cluster or workload
+  * The workload must be demonstrated to use fixed memory or CPU, and the component must have a plan to detect whether the limit causes impact to end user workloads
+  * Workloads that have a dynamic range of usage are not a good candidate for fixed limits.
+  * Memory limits must be set to 25-50% above the observed maximum in standard e2e runs to ensure additional headroom
+  * Examples:
+    * A health check sidecar container that performs a very simple request
+    * A controller that looks at a fixed set of objects that cannot vary based on cluster size, workload scale, or end user action
+
+In general, per-operator dynamic calculation of limits is discouraged, and instead workloads are expected to regulate their own consumption where possible. In cases where the workload has a very large dynamic range, dynamic sharding or scaling using standard (or per operator workload) is to be preferred over dynamic limit setting.
+
+To get approval for an exception, contact a release architect and an exception will be recorded via code in the standard e2e suite which will allow changes to merge. Non-payload operators are encouraged to seek approval as well to better document use cases - other solutions may be suggested or problems in the platform may be uncovered.
```

Comment:
Is there a team page or other reference for "release architect"? If I'm new to the org, how do I find one?

The second sentence here feels like a different train of thought, so I suggest creating a new paragraph for it.

Reply:
Hrm, it's how to get an exception, the second sentence is "who to get it from"

Reply:
I wasn't clear in my feedback.

> To get approval for an exception, contact a release architect and an exception will be recorded via code in the standard e2e suite which will allow changes to merge. 

This sentence is doing a lot of work and could be expanded a bit to provide more context for the reader.

1. It introduces the idea that there is a separate, required process to gain approval for using limits. Is this section of this document meant to be the full description of that process? If so, is there a formal process or is it "start a conversation with a release architect"? If it is more complicated than we want to lay out in full here, is there another doc we could link to?
2. It says that using limits requires review by a specific group of people, rather than normal code review by the team that owns the component being limited. Maybe that's the only "process"? How does someone gain the attention of that group? Who is in the group (how do I find them if I want to talk to them)?
3. It also says that we use a CI job to enforce that most components do not have limits. Is that a check of the YAML files or is there some characteristic of the e2e test suite that prevents limits from working as a side-effect? We may not want to mention which CI job enforces the rule, since the name may change over time, but can we link to a list of CI jobs with their purposes so that someone could find the job definition to see how it works?

> Non-payload operators are encouraged to seek approval as well to better document use cases - other solutions may be suggested or problems in the platform may be uncovered.

This sentence suggests that things that are not part of the release payload should also follow these rules, which explains the scope of the rules discussed above but isn't really part of the process of applying them. Maybe we should talk about the scope at the start of the new section? Something like "OpenShift components, including those in the release payload and those delivered to customers separately, should not have resource limits applied in most circumstances. <updated content from line 306 here>"